components/dashboard/package.json

```diff
@@ -48,6 +50,8 @@
     "start": "craco start",
     "build": "craco build --verbose",
     "test": "craco test",
+    "test:unit": "craco test",
+    "test:e2e": "cypress open",
```

Comment:
For me this is not about e2e testing at this point. I'd rather rename this to `cypress` or so, do indicate that this opens the interactive test runner, which might be very helpful for testing.

Reply:
I added `test:unit` and `test:e2e` as variations since the first one uses Jest for the couple unit tests that are already there (by @laushinka IIRC), while the second one is the new Cypress one.
I wanted to delete the now redundant default "test" script created by Create React App but I didn't want to accidentally break any CI job that might be using it to launch those couple unit tests.

## My proposal

- Delete the "test" script and hope for the best 🤞 about CI (joking 😄 do you know of any such job that might need updating to point at the new "test:unit" script?)
- For unit tests, Craco and CRA run jest in watch mode by default. To avoid that, CI can set a `CI=true` env variable or add `--watchAll=false` to the command line 👌 
- Rename "test:e2e" into "test:integration:watch"
- Add new "test:integration:run" script (`cypress run`) aimed at CI and automation

Reply:
The "test" command can be configured here: https://github.com/gitpod-io/gitpod/blob/28bf0e6541255d43ea1a317ff852ee79b9e14147/components/dashboard/BUILD.yaml#L20-L21
That should be set to `test: ["yarn", "test"]`.

`yarn test` by convention (see other TS components) is the script that executes all "tests that are executed on parts of the code base, do not have component-external dependencies, are non-interactive and have quick turn arounds". This is to enable everyone - without deeper knowledge of the component, or tech involved - to run these _the same way CI does_. I think this is a very nice property worth maintaining. We can discuss for sure, but that's maybe out of scope of the PR.

All other scripts are there to offer more fine grained control; :+1: for your suggestions!

Reply:
✅ scripts updated

I didn't add the test command to BUILD.yaml though, because I noticed `dontTest: true` and I'm not sure what it means. I guess it can be done in a different PR?

Reply:
>  didn't add the test command to BUILD.yaml though, because I noticed dontTest: true and I'm not sure what it means. I guess it can be done in a different PR?

:100: